include/revng-c/mlir/Dialect/Clift/IR/CliftOps.td

```diff
@@ -170,18 +174,15 @@ def Clift_MakeLabelOp : Clift_Op<"make_label",
   ];
 }
 
-def Clift_AssignLabelOp : Clift_Op<"assign_label",
-                                   [Clift_StatementOpInterface]> {
+def Clift_AssignLabelOp : Clift_StatementOp<"assign_label"> {
   let arguments = (ins Clift_LabelType:$label);
 
   let assemblyFormat = [{
     $label type($label) attr-dict
   }];
 }
 
-def Clift_DoWhileOp : Clift_Op<"do_while",
-                               [Clift_StatementOpInterface,
-                                NoTerminator]> {
```

Comment:
Not really related to this PR, since it was there before, but I think we should thing more thoroughly about the `NoTerminator` trait you've added to many operations.

My intuition at the moment, with the current design of the dialect is that:
- we only have single-block regions both in statements and in expressions
- none of these regions ever require the single block to have a terminator
- `ExpressionRegion`s do `verify` that they end with a `YieldOp`

If these are true, and my reasoning is correct, then we basically have that every clift statement and every clift expression could have the `NoTerminator` trait.
At that point we might very well put it in the `StatementOp` (and later in `ExpressionOp`), don't you think?

Reply:
Actually I think we can put `SingleBlock`, `NoRegionArguments` and `NoTerminator` all in `Clift_Op`. These should apply to all of our operations, don't you think?

Reply:
I agree for `SingleBlock` and `NoTerminator`.

For `NoRegionArguments` I think it should be for sure on `StatementOp` and on `ExpressionOp`. I don't have strong opinions about `GlobalVariableOp`. But I don't think it's appropriate on `FunctionOp`, because my understanding is that function arguments should also be region arguments of the body region. Isn't that the case?

Reply:
Yeah, that might be. Turns out `SingleBlock` is not appropriate though, because beyond verifying, it also adds that block to empty regions and we're relying on empty regions in some cases.

Reply:
Yeah, that might be. Turns out `SingleBlock` is not appropriate though, because beyond verifying, it also adds that block to empty regions and we're relying on empty regions in some cases.

Reply:
Oh I see. Can we do `SingleBlockOrEmpty` instead?

Reply:
It doesn't seem necessary. All of our operation regions are either `StatementRegion`s or `ExpressionRegion`s, which do the verification already, except for the `Clift_ModuleOp` region, and in that case `SingleBlock` is appropriate and present.

Reply:
:+1: